File: 882166-Deep_Clone_Function_Fix/evaluation/evaluation.js
const fs = require('fs');
const path = require('path');
const os = require('os');
const { spawnSync } = require('child_process');
const crypto = require('crypto');

const ROOT = path.resolve(__dirname, '..');
const REPORTS_DIR = path.join(ROOT, 'evaluation', 'reports');

function environmentInfo() {
  return {
    node_version: process.version,
    platform: `${os.platform()}-${os.arch()}`
  };
}

function runJestJson(repoName) {
  try {
    const env = { ...process.env, REPO_PATH: repoName };
    // Run the local jest binary via npx so we get JSON output
    const result = spawnSync('npx', ['jest', '--json', '--runInBand', '--testLocationInResults'], {
      cwd: ROOT,
      env,
      encoding: 'utf-8',
      timeout: 120_000
    });

    const raw = (result.stdout || '') + (result.stderr || '');
    let parsed = null;
    try {
      parsed = JSON.parse(result.stdout);
    } catch (e) {
      // Try to find JSON blob inside stdout
      const m = result.stdout.match(/\{[\s\S]*\}\s*$/);
      if (m) {
        try { parsed = JSON.parse(m[0]); } catch (e2) { parsed = null; }
      }
    }

    if (parsed && typeof parsed === 'object') {
      // Build structured output
      const summary = {
        numTotalTests: parsed.numTotalTests || 0,
        numPassedTests: parsed.numPassedTests || 0,
        numFailedTests: parsed.numFailedTests || 0,
        numTotalTestSuites: parsed.numTotalTestSuites || 0,
        numPassedTestSuites: parsed.numPassedTestSuites || 0,
        numFailedTestSuites: parsed.numFailedTestSuites || 0
      };

      const tests = (parsed.testResults || []).flatMap(suite =>
        (suite.assertionResults || []).map(t => ({
          fullName: t.fullName,
          status: t.status,
          title: t.title,
          failureMessages: t.failureMessages || [],
          location: t.location || null
        }))
      );

      return {
        passed: summary.numFailedTests === 0,
        return_code: result.status,
        summary,
        summary_matrix: [[summary.numPassedTests, summary.numFailedTests]],
        tests,
        raw_output: raw.slice(0, 8000)
      };
    }

    // Fallback: no JSON parse
    return {
      passed: result.status === 0,
      return_code: result.status,
      raw_output: raw.slice(0, 8000)
    };
  } catch (err) {
    return {
      passed: false,
      return_code: -1,
      raw_output: err.message
    };
  }
}

function runTests(repoName) {
  // prefer structured jest JSON; if jest unavailable or parse fails, fall back to npm test (human output)
  const structured = runJestJson(repoName);
  if (structured && typeof structured === 'object') return structured;

  // fallback (shouldn't normally be reached)
  try {
    const env = { ...process.env, REPO_PATH: repoName };
    const result = spawnSync('npm', ['test', '--', '--runInBand'], {
      cwd: ROOT,
      env,
      encoding: 'utf-8',
      timeout: 120_000
    });
    return {
      passed: result.status === 0,
      return_code: result.status,
      raw_output: ((result.stdout || '') + (result.stderr || '')).slice(0, 8000)
    };
  } catch (err) {
    return { passed: false, return_code: -1, raw_output: err.message };
  }
}

function runMetrics(_repoName) { return {}; }

function evaluate(repoName) {
  const tests = runTests(repoName);
  return { tests, metrics: runMetrics(repoName) };
}

function runEvaluation() {
  const runId = crypto.randomUUID();
  const start = new Date();

  let before = null, after = null, error = null;
  try {
    before = evaluate('repository_before');
    after = evaluate('repository_after');
  } catch (e) {
    error = e.message;
  }

  const end = new Date();
  const comparison = {
    passed_gate: !!(after && after.tests && after.tests.passed),
    improvement_summary: after && after.tests && after.tests.passed
      ? 'After implementation passed correctness tests'
      : 'After implementation failed correctness tests'
  };

  return {
    run_id: runId,
    started_at: start.toISOString(),
    finished_at: end.toISOString(),
    duration_seconds: (end - start) / 1000,
    environment: environmentInfo(),
    before: before || null,
    after: after || null,
    comparison,
    success: comparison.passed_gate,
    error
  };
}

function main() {
  fs.mkdirSync(REPORTS_DIR, { recursive: true });

  const report = runEvaluation();

  const now = new Date();
  const dateStr = now.toISOString().slice(0, 10); // YYYY-MM-DD
  const timeStr = now.toISOString().slice(11, 19).replace(/:/g, '-'); // HH-MM-SS
  const dirName = `${dateStr}/${timeStr}`;
  const reportDir = path.join(REPORTS_DIR, dirName);
  fs.mkdirSync(reportDir, { recursive: true });

  const reportPath = path.join(reportDir, 'report.json');

  fs.writeFileSync(reportPath, JSON.stringify(report, null, 2));
  console.log(`Report written to ${reportPath}`);

  return report.success ? 0 : 1;
}

if (require.main === module) {
  process.exit(main());
}

File: 830687-separate-squares-problem/evaluation/evaluation.py
#!/usr/bin/env python3
"""
Evaluation runner for Square Split Line Problem.
Runs pytest on both repository_before and repository_after.
"""
import os
import sys
import json
import uuid
import platform
import subprocess
from pathlib import Path
from datetime import datetime

ROOT = Path(__file__).resolve().parent.parent
REPORTS = ROOT / "evaluation" / "reports"


def environment_info():

    """Collect environment information."""
    
    return {
        "python_version": platform.python_version(),
        "platform": platform.platform()
    }


def run_tests(repo_path: Path):
    """Run pytest for a repository and parse detailed test results."""
    import re
    
    env = os.environ.copy()
    env["PYTHONPATH"] = str(repo_path)
    
    try:
        proc = subprocess.run(
            [sys.executable, "-m", "pytest", str(ROOT / "tests"), "-q", "--tb=short"],
            cwd=ROOT,
            capture_output=True,
            text=True,
            timeout=120,
            env=env
        )
        output = (proc.stdout + proc.stderr)[:8000]
        
        # Parse pytest output to extract test counts
        # Look for patterns like "15 failed, 2 passed" or "17 passed"
        passed_count = 0
        failed_count = 0
        total_count = 0
        
        # Try to extract counts from output
        # Pattern 1: "X failed, Y passed" or "X passed, Y failed"
        match = re.search(r'(\d+)\s+failed', output, re.IGNORECASE)
        if match:
            failed_count = int(match.group(1))
        
        match = re.search(r'(\d+)\s+passed', output, re.IGNORECASE)
        if match:
            passed_count = int(match.group(1))
        
        total_count = passed_count + failed_count
        
        # If we couldn't parse, try alternative patterns
        if total_count == 0:
            # Look for "X passed in Y.Ys" pattern
            match = re.search(r'(\d+)\s+passed\s+in', output, re.IGNORECASE)
            if match:
                passed_count = int(match.group(1))
                total_count = passed_count
                # Check return code to see if there were failures
                if proc.returncode != 0:
                    # If return code is non-zero but we only see "passed", 
                    # pytest might have been interrupted or there were errors
                    failed_count = 1 if proc.returncode != 0 else 0
        
        # Determine if tests passed (all tests must pass)
        all_passed = (proc.returncode == 0) and (failed_count == 0) and (total_count > 0)
        
        return {
            "passed": all_passed,
            "return_code": proc.returncode,
            "output": output,
            "test_counts": {
                "total": total_count,
                "passed": passed_count,
                "failed": failed_count
            }
        }
    except subprocess.TimeoutExpired:
        return {
            "passed": False,
            "return_code": -1,
            "output": "pytest timeout",
            "test_counts": {
                "total": 0,
                "passed": 0,
                "failed": 0
            }
        }
    except Exception as e:
        return {
            "passed": False,
            "return_code": -1,
            "output": f"Error running tests: {str(e)}",
            "test_counts": {
                "total": 0,
                "passed": 0,
                "failed": 0
            }
        }


def run_metrics(repo_path: Path):
    return {}


def evaluate(repo_name: str):
    repo_path = ROOT / repo_name
    tests = run_tests(repo_path)
    metrics = run_metrics(repo_path)
    return {
        "tests": tests,
        "metrics": metrics
    }


def run_evaluation():

    """Run the complete evaluation and return report dict."""
    
    run_id = str(uuid.uuid4())
    start = datetime.utcnow()
    
    try:
        before = evaluate("repository_before")
        after = evaluate("repository_after")
        
        before_tests = before["tests"]
        after_tests = after["tests"]
        
        before_passed = before_tests.get("test_counts", {}).get("passed", 0)
        before_failed = before_tests.get("test_counts", {}).get("failed", 0)
        before_total = before_tests.get("test_counts", {}).get("total", 0)
        
        after_passed = after_tests.get("test_counts", {}).get("passed", 0)
        after_failed = after_tests.get("test_counts", {}).get("failed", 0)
        after_total = after_tests.get("test_counts", {}).get("total", 0)
        
        passed_gate = after_tests["passed"] and (after_failed == 0)
        
        # Create detailed improvement summary
        if passed_gate:
            if before_failed > 0:
                improvement_summary = f"After implementation: {after_passed}/{after_total} tests passed (improved from {before_passed}/{before_total} passed, {before_failed} failed in before)"
            else:
                improvement_summary = f"After implementation: {after_passed}/{after_total} tests passed"
        else:
            if after_failed > 0:
                improvement_summary = f"After implementation: {after_failed} failed, {after_passed} passed out of {after_total} total tests"
            else:
                improvement_summary = "After implementation failed correctness tests"
        
        end = datetime.utcnow()
        
        return {
            "run_id": run_id,
            "started_at": start.isoformat() + "Z",
            "finished_at": end.isoformat() + "Z",
            "duration_seconds": (end - start).total_seconds(),
            "environment": environment_info(),
            "before": before,
            "after": after,
            "comparison": {
                "passed_gate": passed_gate,
                "improvement_summary": improvement_summary,
                "before_test_summary": f"{before_passed} passed, {before_failed} failed out of {before_total} total",
                "after_test_summary": f"{after_passed} passed, {after_failed} failed out of {after_total} total"
            },
            "success": passed_gate,
            "error": None
        }
    except Exception as e:
        end = datetime.utcnow()
        return {
            "run_id": run_id,
            "started_at": start.isoformat() + "Z",
            "finished_at": end.isoformat() + "Z",
            "duration_seconds": (end - start).total_seconds(),
            "environment": environment_info(),
            "before": {
                "tests": {
                    "passed": False,
                    "return_code": -1,
                    "output": "",
                    "test_counts": {
                        "total": 0,
                        "passed": 0,
                        "failed": 0
                    }
                },
                "metrics": {}
            },
            "after": {
                "tests": {
                    "passed": False,
                    "return_code": -1,
                    "output": "",
                    "test_counts": {
                        "total": 0,
                        "passed": 0,
                        "failed": 0
                    }
                },
                "metrics": {}
            },
            "comparison": {
                "passed_gate": False,
                "improvement_summary": "Evaluation crashed"
            },
            "success": False,
            "error": str(e)
        }


def main():
    """Main entry point."""
    REPORTS.mkdir(parents=True, exist_ok=True)
    
    try:
        report = run_evaluation()
        latest_path = REPORTS / "latest.json"
        report_path = REPORTS / "report.json"
        
        report_json = json.dumps(report, indent=2)
        latest_path.write_text(report_json)
        report_path.write_text(report_json)
        
        print(f"Report written to {latest_path}")
        print(f"Report written to {report_path}")
        return 0 if report["success"] else 1
    except Exception as e:
        print(f"Fatal error: {e}", file=sys.stderr)
        return 1


if __name__ == "__main__":
    sys.exit(main())
